### DISCUSSION TITLE
Video encoding with html5-client?

### DISCUSSION THREAD

**Discussion by skipperro** [Xpra-org/xpra#4422] [Q&A]
I would like to use Xpra as fast remote desktop for my docker containers, accessible via web browser. 
I managed to make almost everything work - I have a container that can use my NVIDIA GPU and it runs xpra-server that I can connect with using desktop xpra client using VP9 and H264 encoding running via NVENC. Everything is good there.

But when I use HTML5 client through web browser it is always switching to using picture encoding instead of video/h264/nvenc.
It even says in the server logs, that only picture encoding is available:
`automatic picture encoding enabled, also available:
rgb24, rgb32, png, webp, jpeg`

So my question is - is video encoding even supported on html5 client?
And if yes, could you point me into right direction to enable it?

**Comment by totaam** [comment]
The problem comes from the html5 client.
There were some bugs - now fixed, but perhaps not all have been included in a release yet.
Then there are still more bugs to do with offscreen decoding: https://github.com/Xpra-org/xpra-html5/issues/329

Because video decoding is quite tricky in browsers, it may not be enabled by default on your browser + setup (ie: are you using https?)
To enable video, the option must be set (advanced connection options) and the html5 client will still validate the encodings on start to verify that they can actually be used. More details should be available in your Javascript console output.
Any of these steps can go wrong and default back to picture encodings.
